src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3299,7 +3290,15 @@ void ApplicationManagerImpl::UnregisterApplication(
       resume_controller().RemoveApplicationFromSaved(app_to_remove);
     }
 
-    subscribed_way_points_apps_list_.erase(app_id);
+    if (IsAppSubscribedForWayPoints(app_id)) {
```

Comment:
Please describe the reason of this change in commit message

Reply:
Done in 144c7a25ae4ce0687856017cd52f94949be370a0